hbase-client/src/main/java/org/apache/hadoop/hbase/client/RpcRetryingCallerFactory.java

```diff
@@ -71,30 +71,41 @@ public <T> RpcRetryingCaller<T> newCaller() {
       interceptor, startLogErrorsCnt, connectionConf.getRpcTimeout(), metrics);
   }
 
-  public static RpcRetryingCallerFactory instantiate(Configuration configuration,
+  @RestrictedApi(explanation = "Should only be called on process initialization", link = "",
```

Comment:
Avoid calling this method as much as is reasonable.

The `HRegionServer` call site is particularly insidious. It is called from the `HRegionServer` constructor and looks like this,

```java
      rpcRetryingCallerFactory = RpcRetryingCallerFactory.instantiate(this.conf,
        clusterConnection == null ? null : clusterConnection.getConnectionMetrics());
```

I think that `clusterConnection` is always null here, because `clusterConnection` is instantiated in `protected synchronized void setupClusterConnection()`, which, as far as I can tell, is only called from `public void run()`.